src/types/User.ts

```diff
@@ -0,0 +1,11 @@
+export interface User {
+    upn: string;
+    name: string;
+    favoriteAds?: number[];
```

Comment:
Why `favoriteAds?` and not `favoriteAds`?

Array can be of length 0...

Reply:
but the ? is not for an empty array it means that favoriteAds is optional

Reply:
True, but he might receive from SQL rows at length 0. Better check if favoriteAds's length is > 0 than check if has been passed (Just like we do with ads).

Reply:
old code. not relevant